### DISCUSSION TITLE
Add a function that when a node is dragged into a script, the content can be filled and modified adaptively, rather than just dropped in

### DISCUSSION THREAD

**Discussion by LingkongSky** [godotengine/godot-proposals#9310] [Editor]
### Describe the project you are working on

![图片](https://github.com/godotengine/godot-proposals/assets/61834524/eb6f6c0b-3d54-4bcd-adc6-e5c24cf201c8)
to
![图片](https://github.com/godotengine/godot-proposals/assets/61834524/7cdbb633-e4a0-4366-8a63-87122b0a552b)

### Describe the problem or limitation you are having in your project

It slows down the coding efficiency to a certain extent

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It can improve the efficiency of developers to a certain extent

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The adaptive filling form can be modified by determining the string structure or regular pattern of the dragging position

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I think it may going to be used a lot

### Is there a reason why this should be core and not an add-on in the asset library?

Because this is necessary to improve the daily use of a game engine

**Comment by AThousandShips** [comment]
Converted to a discussion as there's no concrete ideas for how this would be done, or even details on what would be included

**Reply by ghost** [reply]
I think it is straightforward. If user drags scene from file dock to editor it just puts scene path to the place where user released the mouse key. If user dragged the scene over string it will just insert the scene path inside the string like in first image. The issue is about ability to replace the string (or resource path) if user dragged it over string in the code

## Current behavior

```
e = event
if e.is_dragged():
    if e.dragged_object.type == scene_file:
        if e.dragged_from == filesystem_dock && e.dragged_to == editor_dock:
            e.editor.insert_to_current_pos(e.dragged_object.get_resource_path())
```

## Desired behavior

```
e = event
if e.is_dragged():
    if e.dragged_object.type == scene_file:
        if e.dragged_from == filesystem_dock && e.dragged_to == editor_dock:
            editor = e.editor
            if editor.pos is inside string in editor (which also could if it is a resource path)
                e.editor.replace_string_where_was_dragged_scene(e.dragged_object.get_resource_path())
```

**Reply by AThousandShips** [reply]
That sounds very specific, why would you drag it like that? If you need to replace some string why not just paste it instead, I haven't used any software that does what you suggest

**Reply by ghost** [reply]
If you've already have one resource string and want to replace it you will need to do other actions. I agree that it is very specific but it is about usability of mouse when you just drag from dock another file and it just replace previous.  Consider the situation. You used one name and decided to move your file to another place. Your script isn't automatically updated to reflect new filesystem structure... Or when you just accidentally dragged `.gd` file instead of desired `.tscn`... Yeah it is very specific but it can be annoying

**Reply by KoBeWi** [reply]
You can *somewhat* achieve this by selecting the old text before dropping.
